### ISSUE TITLE
LOOT 0.19.1 fails to build with a TBB error

### ISSUE THREAD

**Issue by C0rn3j** [loot/loot#1864]
[build.log](https://github.com/loot/loot/files/11469644/build.log)

Using 
```
onetbb 2021.8.0
loot 0.19.1
libloot 0.19.4
```

This is on Arch Linux building on Ryzen 5 7600X.

Excerpt from the log:
```yaml
/usr/include/oneapi/tbb/profiling.h:229:15: error: expected unqualified-id before ‘)’ token
  229 |     void emit() { }
      |               ^
```

I am not 100% whether this is an issue with LOOT or an issue with something else...

**Comment by Ortham** [comment]
I suspect that this is because Qt has an `emit` macro, and that's clashing with the function in tbb. I'm not sure how to solve this though: usually I'd `#undef` the macro before including the library, but LOOT doesn't use tbb directly, it's a standard library dependency, and I've got tbb 2020.1-2 installed on my Ubuntu VM.

Maybe try adding `#undef emit` before the `#include <execution>` line in `gui/state/game/game.h` and `#define emit` afterwards? It'll probably need the `#define` too since it's in a header file that's included by files that use the emit macro.

I.e. replace

```c++
#include <execution>
#include <filesystem>
#include <functional>
#include <mutex>
#include <optional>
#include <string>
#include <variant>
```

with 

```c++
#undef emit

#include <execution>
#include <filesystem>
#include <functional>
#include <mutex>
#include <optional>
#include <string>
#include <variant>

#define emit
```

**Comment by C0rn3j** [comment]
That does the trick actually!

![image](https://github.com/loot/loot/assets/1641362/5b3fee04-cb57-4b80-b3aa-2843b86f344d)

**Comment by Ortham** [comment]
I've just pushed a `fix-linux-build` branch that conditionally does the undef-then-define, can you verify it works as expected?

**Comment by C0rn3j** [comment]
I didn't try the branch directly, but I took your commit from it and applied it against the stable release version and it does indeed work fine for me

**Comment by Ortham** [comment]
Thanks, that means this is fixed in d5a3dc94f5b503ca9023220f85dfea50b409b014.
